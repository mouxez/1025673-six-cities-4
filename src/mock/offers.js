const offers = [
  {
    id: 1,
    isPremium: true,
    imgSrc: `http://picsum.photos/260/200?r=1`,
    imgDescription: `Some photo description`,
    price: 100,
    isBookmarked: false,
    placeDescription: `Some place description`,
    placeType: `Apartment`,
  },
  {
    id: 2,
    isPremium: false,
    imgSrc: `http://picsum.photos/260/200?r=2`,
    imgDescription: `Some photo description`,
    price: 70,
    isBookmarked: true,
    placeDescription: `Some place description`,
    placeType: `Apartment`,
  },
  {
    id: 3,
    isPremium: false,
    imgSrc: `http://picsum.photos/260/200?r=3`,
    imgDescription: `Some photo description`,
    price: 90,
    isBookmarked: true,
    placeDescription: `Some place description`,
    placeType: `Room`,
  },
  {
    id: 4,
    isPremium: true,
    imgSrc: `http://picsum.photos/260/200?r=4`,
    imgDescription: `Some photo description`,
    price: 120,
    isBookmarked: false,
    placeDescription: `Some place description`,
    placeType: `House`,
  },
  {
    id: 5,
    isPremium: true,
    imgSrc: `http://picsum.photos/260/200?r=5`,
    imgDescription: `Some photo description`,
    price: 40,
    isBookmarked: false,
    placeDescription: `Some place description`,
    placeType: `Hotel`,
  },
];

export {offers};
