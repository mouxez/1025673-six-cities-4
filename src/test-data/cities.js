const cities = [{location: [59.3198, 18.0601], name: `Stockholm`}];

export {cities};
