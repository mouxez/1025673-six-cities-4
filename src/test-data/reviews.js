const reviews = [
  {
    "avatarUrl": `http://picsum.photos/90/90?r=1`,
    "user": `John`,
    "rating": 3.3,
    "description": `A quiet cozy and picturesque that hides behind a river by the unique lightness of Amsterdam.
      The building is green and from 18th century.`,
    "date": `2020-07-02`,
  }
];

export {reviews};
