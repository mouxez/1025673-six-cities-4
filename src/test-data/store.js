import configureStore from 'redux-mock-store';
import {cities} from './cities';
import {offers} from './offers';

const mockStore = configureStore([]);

const store = mockStore({
  activeCityName: cities[0].name,
  offers,
  cities,
});

export {store};
