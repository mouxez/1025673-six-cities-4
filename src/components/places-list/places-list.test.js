import React from 'react';
import renderer from 'react-test-renderer';
import {offers} from '../../test-data/offers';

import PlacesList from './places-list';

test(`PlacesList should render correctlt`, () => {
  const tree = renderer.create(
      <PlacesList
        offers={offers}
        onPlaceCardTitleClick={jest.fn()}
      />).toJSON();

  expect(tree).toMatchSnapshot();
});
