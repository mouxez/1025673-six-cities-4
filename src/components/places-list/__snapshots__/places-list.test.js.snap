// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlacesList should render correctlt 1`] = `
<div
  className="places__list cities__places-list tabs__content"
>
  <article
    className="place-card cities__place-card"
    onMouseEnter={[Function]}
  >
    
    <div
      className="place-card__image-wrapper cities__image-wrapper"
    >
      <a
        href="/"
      >
        <img
          alt="property image"
          className="place-card__image"
          height="200"
          src="http://picsum.photos/260/200?r=1"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            110
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "96%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            4.8
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Beautiful & luxurious studio at great location
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>
  <article
    className="place-card cities__place-card"
    onMouseEnter={[Function]}
  >
    
    <div
      className="place-card__image-wrapper cities__image-wrapper"
    >
      <a
        href="/"
      >
        <img
          alt="property image"
          className="place-card__image"
          height="200"
          src="http://picsum.photos/260/200?r=2"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            80
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "56%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            2.8
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Beautiful & luxurious studio at great location
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        room
      </p>
    </div>
  </article>
  <article
    className="place-card cities__place-card"
    onMouseEnter={[Function]}
  >
    
    <div
      className="place-card__image-wrapper cities__image-wrapper"
    >
      <a
        href="/"
      >
        <img
          alt="property image"
          className="place-card__image"
          height="200"
          src="http://picsum.photos/260/200?r=3"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            20
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "76%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            3.8
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Beautiful & luxurious studio at great location
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        house
      </p>
    </div>
  </article>
  <article
    className="place-card cities__place-card"
    onMouseEnter={[Function]}
  >
    
    <div
      className="place-card__image-wrapper cities__image-wrapper"
    >
      <a
        href="/"
      >
        <img
          alt="property image"
          className="place-card__image"
          height="200"
          src="http://picsum.photos/260/200?r=4"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            100
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            5
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Beautiful & luxurious studio at great location
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        hotel
      </p>
    </div>
  </article>
  <article
    className="place-card cities__place-card"
    onMouseEnter={[Function]}
  >
    
    <div
      className="place-card__image-wrapper cities__image-wrapper"
    >
      <a
        href="/"
      >
        <img
          alt="property image"
          className="place-card__image"
          height="200"
          src="http://picsum.photos/260/200?r=5"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            130
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "48%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            2.4
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Beautiful & luxurious studio at great location
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        hotel
      </p>
    </div>
  </article>
  <article
    className="place-card cities__place-card"
    onMouseEnter={[Function]}
  >
    
    <div
      className="place-card__image-wrapper cities__image-wrapper"
    >
      <a
        href="/"
      >
        <img
          alt="property image"
          className="place-card__image"
          height="200"
          src="http://picsum.photos/260/200?r=6"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            160
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "70%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            3.5
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Beautiful & luxurious studio at great location
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        hotel
      </p>
    </div>
  </article>
</div>
`;
